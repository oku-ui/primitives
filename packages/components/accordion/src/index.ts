export { OkuAccordion } from './accordion'

export {
  type AccordionMultipleEmits, accordionProps, type AccordionMultipleProps, type AccordionSingleEmits, type AccordionSingleProps,
  accordionContentProps,
  type AccordionContentProps, accordionHeaderProps,
  type AccordionHeaderProps, accordionImplProps,
  type AccordionImplProps, type AccordionImplEmits,
  accordionImplSingleProps,
  type AccordionImplSingleProps, type AccordionImplSingleEmits,
  accordionImplMultipleProps,
  type AccordionImplMultipleProps, type AccordionImplMultipleEmits,
  accordionItemProps,
  type AccordionItemProps, type AccordionItemEmits,
  accordionTriggerProps,
  type AccordionTriggerProps, type AccordionTriggerEmits,
} from './props'
export {
  OkuAccordionContent,
} from './accordionContent'

export {
  OkuAccordionHeader,
} from './accordionHeader'

export {
  OkuAccordionImpl,
} from './accordionImpl'

export {
  OkuAccordionImplSingle,
} from './accordionImplSingle'

export {
  OkuAccordionImplMultiple,
} from './accordionImplMultiple'

export {
  OkuAccordionItem,
} from './accordionItem'

export {
  OkuAccordionTrigger,
} from './accordionTrigger'
