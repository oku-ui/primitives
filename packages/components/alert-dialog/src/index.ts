export {
  OkuAlertDialog,
} from './alert-dialog'

export {
  OkuAlertDialogAction,
} from './alert-dialog-action'

export {
  OkuAlertDialogCancel,
} from './alert-dialog-cancel'

export {
  OkuAlertDialogContent,
} from './alert-dialog-content'

export {
  OkuAlertDialogDescription, alertDialogDescriptionProps,
} from './alert-dialog-description'

export {
  OkuAlertDialogDescriptionWarning, dialogDescriptionWarningProps,
} from './alert-dialog-description-warning'

export {
  OkuAlertDialogOverlay, alertDialogOverlayProps,
} from './alert-dialog-overlay'

export {
  OkuAlertDialogPortal, alertDialogPortalProps,
} from './alert-dialog-portal'

export {
  OkuAlertDialogTitle, alertDialogTitleProps,
} from './alert-dialog-title'

export {
  OkuAlertDialogTrigger, alertDialogTriggerProps,
} from './alert-dialog-trigger'

export type {
  AlertDialogProps, AlertDialogActionProps, AlertDialogCancelProps, AlertDialogContentProps, AlertDialogDescriptionProps, AlertDialogOverlayProps, AlertDialogPortalProps, AlertDialogTitleProps, AlertDialogTriggerProps, DescriptionWarningProps,
} from './props'

export {
  alertDialogProps,
  alertDialogActionProps,
  alertDialogCancelProps,
  alertDialogContentProps,
} from './props'
