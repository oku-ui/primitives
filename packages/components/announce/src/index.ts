export {
  OkuAnnounce,
} from './announce'

export type { AnnounceProps, AnnounceElement } from './announce'
