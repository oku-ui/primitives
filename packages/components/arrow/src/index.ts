export {
  OkuArrow,
  arrowProps,
} from './arrow'

export type {
  ArrowProps,
  ArrowElement,
  ArrowIntrinsicElement,
} from './arrow'
