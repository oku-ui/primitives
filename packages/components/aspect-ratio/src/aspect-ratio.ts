import type { PropType } from 'vue'
import { defineComponent, h, toRef } from 'vue'
import type { OkuElement, PrimitiveProps } from '@oku-ui/primitive'
import { Primitive, primitiveProps } from '@oku-ui/primitive'
import { useForwardRef } from '@oku-ui/use-composable'

export type AspectRatioNaviteElement = OkuElement<'div'>
export type AspectRatioElement = HTMLDivElement

const NAME = 'OkuAspectRatio'

export interface AspectRatioProps extends PrimitiveProps {
  ratio?: number
}

export const aspectRatioProps = {
  props: {
    ratio: {
      type: Number as PropType<number>,
      default: 1 / 1,
    },
  },
  emits: {},
}

const aspectRatio = defineComponent({
  name: NAME,
  inheritAttrs: false,
  props: {
    ...aspectRatioProps.props,
    ...primitiveProps,
  },
  setup(props, { attrs, slots }) {
    const ratio = toRef(props, 'ratio')

    const forwardedRef = useForwardRef()
    const originalReturn = () => h(
      'div',
      {
        'style': {
          position: 'relative',
          width: '100%',
          paddingBottom: `${100 / ratio.value}%`,
        },
        'data-oku-aspect-ratio-wrapper': '',
      },
      [
        h(
          Primitive.div,
          {
            asChild: props.asChild,
            ...attrs,
            ref: forwardedRef,
            style: {
              ...attrs.style as any,
              position: 'absolute',
              top: '0px',
              right: '0px',
              left: '0px',
              bottom: '0px',
            },
          },
          {
            default: () => slots.default?.(),
          },
        ),
      ],
    )

    return originalReturn
  },
})

// TODO: https://github.com/vuejs/core/pull/7444 after delete
export const OkuAspectRatio = aspectRatio as typeof aspectRatio &
(new () => {
  $props: AspectRatioNaviteElement
})
