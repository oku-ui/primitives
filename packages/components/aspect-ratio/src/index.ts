export { OkuAspectRatio } from './aspect-ratio'
export type {
  AspectRatioProps,
  AspectRatioElement,
  AspectRatioIntrinsicElement,
} from './aspect-ratio'
