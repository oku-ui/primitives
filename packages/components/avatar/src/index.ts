export {
  OkuAvatar,
  createAvatarScope,
  avatarProps,
} from './avatar'

export {
  OkuAvatarFallback,
  avatarFallbackProps,
} from './avatarFallback'

export {
  OkuAvatarImage,
  avatarImageProps,
} from './avatarImage'

export type {
  AvatarImageProps,
  AvatarImageElement,
  AvatarImageIntrinsicElement,
  AvatarEmits,
} from './avatarImage'

export type {
  AvatarProps,
  AvatarElement,
  AvatarIntrinsicElement,
} from './avatar'

export type {
  AvatarFallbackProps,
  AvatarFalbackElement,
  AvatarFallbackIntrinsicElement,
} from './avatarFallback'
