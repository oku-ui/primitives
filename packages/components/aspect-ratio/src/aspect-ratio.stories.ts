import type { Meta, StoryObj } from '@storybook/vue3'

import type { AspectRatioProps } from './aspect-ratio'
import { OkuAspectRatio } from './aspect-ratio'

interface StoryProps extends AspectRatioProps {
  imageurl: string
}

const meta = {
  title: 'Components/OkuAspectRatio',
  component: OkuAspectRatio,
  tags: ['autodocs'],
  argTypes: {
    // TODO: ratio number '16 / 9' not working how to send a string? with storybook controls
    ratio: { type: 'number', defaultValue: '1' },
  },
  // TODO: ratio number '16 / 9' not working how to send a string? with storybook controls
  args: { ratio: '1' },
  // TODO: `render` ts props same problem as above
  render: (args: StoryProps) => ({
    components: { OkuAspectRatio },
    setup() {
      return { args }
    },
    // TODO: if ratio hot reload fixed, remove this {{ args.ratio }}
    // TODO: ratio change not working hot reload
    template: `
      {{ args.ratio }}
      <div class="max-w-xl mx-auto h-full items-center justify-center">
        <OkuAspectRatio :ratio="args.ratio">
          <img
            class="object-cover w-full h-full rounded-lg"
            :src="args.imageurl"
            alt="Landscape photograph by Tobias Tullius"
          >
        </OkuAspectRatio>
      </div>
    `,
  }),
} satisfies Meta<typeof OkuAspectRatio>

export default meta
type Story = StoryObj<typeof meta>

export const Primary: Story = {
  args: {
    ratio: '1',
    imageurl: 'https://images.unsplash.com/photo-1535025183041-0991a977e25b?w=300&dpr=2&q=80',
  },
}
