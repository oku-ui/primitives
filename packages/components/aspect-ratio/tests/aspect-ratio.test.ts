import { enableAutoUnmount, mount } from '@vue/test-utils'
import { afterEach, describe, expect, it } from 'vitest'
import type { Component } from 'vue'
import { h } from 'vue'
import { axe } from 'vitest-axe'
import { OkuAspectRatio } from '../src'

const component = {
  setup(props, { attrs, slots }) {
    return () => h(OkuAspectRatio, { ...attrs }, slots)
  },
} as Component

describe('OkuAspectRatio', () => {
  enableAutoUnmount(afterEach)

  it('renders the component correctly', () => {
    const wrapper = mount(component)
    expect(wrapper.exists()).toBe(true)
  })

  /**
  * @vitest-environment jsdom
  */
  it('should have no accessibility violations', async () => {
    const _wrapper = mount(component)
    const results = await axe(_wrapper.element)
    // https:// github.com/capricorn86/happy-dom/issues/978
    // TODO:77 https://github.com/chaance/vitest-axe/issues/7
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-expect-error
    expect(results).toHaveNoViolations()
  })

  it('component render', async () => {
    const wrapper = mount(component)
    expect(wrapper.element).toMatchSnapshot()
  })
})
