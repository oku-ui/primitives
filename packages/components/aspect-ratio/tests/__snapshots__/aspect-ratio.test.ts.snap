// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OkuAspectRatio > component render 1`] = `
<div
  data-oku-aspect-ratio-wrapper=""
  style="position: relative; width: 100%; padding-bottom: 100%;"
>
  <div
    style="position: absolute; top: 0px; right: 0px; left: 0px; bottom: 0px;"
  >
    <!---->
  </div>
</div>
`;

exports[`okuAspectRatio > component render 1`] = `
<div
  data-oku-aspect-ratio-wrapper=""
  style="position: relative; width: 100%; padding-bottom: 100%;"
>
  <div
    style="position: absolute; top: 0px; right: 0px; left: 0px; bottom: 0px;"
  >
    <!---->
  </div>
</div>
`;
